go.mod

```diff
@@ -10,7 +10,7 @@ require (
 	github.com/giantswarm/apiextensions v0.0.0-20191213075442-71155aa0f5b7
 	github.com/giantswarm/columnize v2.0.3-0.20190718092621-cc99d98ffb29+incompatible
 	github.com/giantswarm/gscliauth v0.2.1
-	github.com/giantswarm/gsclientgen v0.0.0-20200423130825-3202ff18d14f
+	github.com/giantswarm/gsclientgen v1.0.2-0.20200507100632-8ff496037482
```

Comment:
Reminder to use a properly tagged version here.

Reply:
This is the result of `go get github.com/giantswarm/gsclientgen@8ff4960` (which is also tagged `v2.0.4`)

Running the command above with `@v2.0.4` fails with: `go get github.com/giantswarm/gsclientgen@v2.0.4: github.com/giantswarm/gsclientgen@v2.0.4: invalid version: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v2`

I think I need a little pointer on how to resolve this

Reply:
I believe you need to use `github.com/giantswarm/gsclientgen/v2` in the code to be able to have the v2 version in go.mod

Reply:
Tried that. Does not work. I've replaced all `github.com/giantswarm/gsclientgen` with `github.com/giantswarm/gsclientgen/v2` in all `*.go` files, then ran `go run main.go`.

Errors with `... module github.com/giantswarm/gsclientgen@latest found (v2.0.3+incompatible), but does not contain package github.com/giantswarm/gsclientgen/v2/client`

Does the [go.mod](https://github.com/giantswarm/gsclientgen/blob/master/go.mod) file of gsclientgen need to specify the same import path? `github.com/giantswarm/gsclientgen/v2` ?

Reply:
https://github.com/giantswarm/gsclientgen/blob/8ff49603748206bdea4ab8f548d7702557a141b2/go.mod#L1

This should be `module github.com/giantswarm/gsclientgen/v2` AFAIK

Reply:
Tried that (https://github.com/giantswarm/gsclientgen/commit/780706a7448880f905e85c0f2610f2df8176b6f1), but with the changed import paths in all files, parameters etc are not compatible any more. I'm guessing all import paths in go files of gsclientgen need to be adapted? This seems like an antipattern since gsclientgen is generated.

Or is there something I am missing?

Reply:
Yeah, you're right. We need to find a way to add versioning to the client generation, or else we will not be able to use released versions

Reply:
So for now If its okay for you @MarcelMue @axbarsan lets keep this like it is and address the changes in gsclientgen separate?

Reply:
Fine for now but yeah let's take care.